Automattic/jetpack
# Blogging Prompts: Add the blogging prompt key to be stored with the post

#### Changes proposed in this Pull Request:
* This PR stores the post_meta info that is related to the blogging prompt that gets answered added to rest requests when the user answers the blogging prompt via the Mobile App. 

#### Jetpack product discussion
See Blogging prompt project. 
pctCYC-kh-p2

Related .com changes. 
D81543-code

#### Does this pull request change what data or activity we track or use?
This changes stores the `_jetpack_blogging_prompt_key` post meta that gets used by the blogging prompt mobile app feature.

#### Testing instructions:
1. Apply this patch to your jetpack sandbox. 
2. Make a request to the site containg the post meta key. `_jetpack_blogging_prompt_key` using the 
[/rest/v1.1/sites/example.com/post/new endpoint
](https://public-api.wordpress.com/rest/v1.1/sites/bitter-bobolink.jurassic.ninja/posts/new)

example body:
```
{ 
  "date": "2022-05-02T14:30:26+0000",
  "terms_by_id": {
    "category": []
  },
  "password": "",
  "terms": {
    "post_tag": []
  },
  "format": "standard",
  "sticky": false,
  "title": "new post 651111 2222",
  "excerpt": "",
  "featured_image": "",
  "content": "123\n<p>testing</p>\n",
  "slug": "",
  "status": "publish",
  "metadata": [
    {
    	"key": "_jetpack_blogging_prompt_key",
      	"value": 1847,
      	"operation": "update"
    }
  ]
}
```
3. Note that the response contains the key. 
3. Note that the `_jetpack_blogging_prompt_key` key gets stored. 
5. Note that a sync request makes it to .com containing the  blogging prompt key
